src/errorReporter.ts

```diff
@@ -0,0 +1,68 @@
+
+import * as vscode from 'vscode';
+
+export const enum ErrorCodes {
+  Error,
+  Warning,
+  Informational
+}
+
+export interface ErrorEventInterface {
+  title: string,
+  message: string | unknown,
+  errorCode: ErrorCodes
+}
+
+export class ErrorReporter {
+  static async showErrorDialog(
+    error: ErrorEventInterface): Promise<void> {
+
+    const template = `<!-- Please delete any private information -->
+        **Version:**
+        ${// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+          vscode.extensions.getExtension('ms-edgedevtools.vscode-edge-devtools')?.packageJSON.version || 'N/A'}
+
+        **OS:**
+        <!-- Add your hosting platform e.g Microsoft Windows, WSL -->
+
+        **Stack :**
+        ${error.message}
+
+        **Additional context:**
+        <!-- Add any other context or about the problem or screenshots here. -->`;
+
+    // cannot do multiline due to:
+    // https://github.com/Microsoft/vscode/issues/48900
+    const answer = await vscode.window
+      .showErrorMessage(
+        `An error occurred:\n${error.title}\n${error.message}`,
+        ...['File a bug', 'Close dialog']
```

Comment:
Nit: I think Olivia brought this up as well, wdyt about removing the Close dialog, since there is already a big x button